packages/common/types.ts

```diff
@@ -192,7 +192,7 @@ export type Requirement2 =
   | IOrCourse2
   | ICourseRange2
   | IRequiredCourse
-  | Section;
+  | ({ type: "SECTION" } & Section);
```

Comment:
I recall Alex and myself discussing this, but it would have changed a lot of other typings used in the rest of the code. Since this is only really required in the validation, we decided to append it here vs change it overall. Thoughts @bugsalexander?

Reply:
pasting convo from slack
![Screen Shot 2022-03-30 at 1 57 03 AM](https://user-images.githubusercontent.com/22948432/160761399-9a726bdd-78f3-4e6e-998e-a8c236c8b728.png)

@BrandonLim8890 on second thought i dont actually think it'd require other places to change in code because we don't use major2 yet anywhere, my bad.

i think cindy's point is probably that it makes more sense to just _have_ it on the section and have the API be more uniform, rather than try and be clever for a reason that might not be clear. does that make sense?

and yeet @cindy1u0 im down to add it in